circus/pidfile.py

```diff
@@ -1,45 +1,51 @@
 import errno
 import os
+import stat
 import tempfile
 
 
 class Pidfile(object):
     """
     Manage a PID file. If a specific name is provided
     it and '"%s.oldpid" % name' will be used. Otherwise
-    we create a temp file using os.mkstemp.
+    we create a temp file using tempfile.mkstemp.
     """
 
     def __init__(self, fname):
         self.fname = fname
-        self.pid = None
+        self.pid = os.getpid()
```

Comment:
`os.gitpid` should be a fairly cheap call, and ensures `Pidfile` always has a valid pid, no matter what it get set to later.